frontends/common/resolveReferences/resolveReferences.h

```diff
@@ -70,6 +76,12 @@ class ResolutionContext : virtual public Visitor, public DeclarationLookup {
 
     const IR::IDeclaration *getDeclaration(const IR::Path *path, bool notNull = false) const;
     const IR::IDeclaration *getDeclaration(const IR::This *, bool notNull = false) const;
+
+    Util::Enumerator<const IR::IDeclaration *> *getDeclarations(const IR::INamespace *ns) const;
+    std::vector<const IR::IDeclaration *> *memoizeDeclarations(const IR::INamespace *ns) const;
```

Comment:
If these functions modify the maps they shouldn't be const, no?

Reply:
No, due to `mutable` used on maps. Semantically these are const functions